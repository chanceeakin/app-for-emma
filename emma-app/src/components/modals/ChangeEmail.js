// @flow
import React, { Component } from 'react'
import {
  Modal,
  Text,
  TouchableHighlight,
  View,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Keyboard,
} from 'react-native'
import GradientWrapper from './../wrappers/GradientWrapper'
import {
  colors,
  button,
  modal as modalStyle,
  pageLayout,
  forms,
} from './../../styles'
import type { SettingsAction } from './../../actions/Settings.js.flow'

const {
  bigButton,
  bigButtonText,
  settingsButton,
  settingsButtonText,
  mediumButton,
  mediumButtonText,
  errorButton,
  errorText,
  backButton,
} = button
const { modal, modalTitle } = modalStyle
const { input } = forms
const { container } = pageLayout

const styles = StyleSheet.create({
  container,
  modal,
  modalTitle,
  bigButton,
  bigButtonText,
  settingsButton,
  settingsButtonText,
  mediumButton,
  mediumButtonText,
  input,
  errorButton,
  errorText,
  backButton,
})

type Props = {
  updatedEmail: string,
  updateEmailPatchTextField: SettingsAction,
  isPatchingEmailError?: boolean,
  changeEmail: () => void,
  isEmailModalShown: boolean,
  toggleEmailModal: SettingsAction,
};

export default class ModalComponent extends Component<Props> {
  render() {
    const {
      updateEmailPatchTextField,
      isPatchingEmailError,
      updatedEmail,
      changeEmail,
      isEmailModalShown,
      toggleEmailModal,
    } = this.props
    return (
      <View style={styles.container}>
        <Modal
          animationType="fade"
          transparent={false}
          visible={isEmailModalShown}
          onRequestClose={() => {
            alert('Modal has been closed.')
          }}
        >
          <GradientWrapper color="orange" style={styles.modal}>
            <Text style={styles.modalTitle}>Change Email</Text>
            <View>
              <TextInput
                style={styles.input}
                onChangeText={text => updateEmailPatchTextField(text)}
                value={updatedEmail}
                placeholder="New Email"
                placeholderTextColor={colors.inkBlue}
                autoCapitalize="none"
                onBlur={Keyboard.dismiss}
              />
              <TouchableOpacity
                style={[
                  styles.bigButton,
                  isPatchingEmailError ? styles.errorButton : null,
                ]}
                onPress={changeEmail}
              >
                <Text
                  style={[
                    styles.bigButtonText,
                    isPatchingEmailError ? styles.errorText : null,
                  ]}
                >
                  Change Email
                </Text>
              </TouchableOpacity>
            </View>
            <TouchableHighlight
              style={styles.backButton}
              onPress={toggleEmailModal}
            >
              <Text>Back</Text>
            </TouchableHighlight>
          </GradientWrapper>
        </Modal>

        <TouchableHighlight
          style={styles.settingsButton}
          onPress={toggleEmailModal}
        >
          <Text style={styles.settingsButtonText}>Change Email</Text>
        </TouchableHighlight>
      </View>
    )
  }
}
