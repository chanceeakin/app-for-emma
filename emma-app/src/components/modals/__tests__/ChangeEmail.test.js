import React from 'react'
import ChangeEmail from '../ChangeEmail'

import renderer from 'react-test-renderer'

const props = {
  updatedEmail: 'email',
  updateEmailPatchTextField: jest.fn(),
  isPatchingEmailError: false,
  changeEmail: jest.fn(),
  isEmailModalShown: false,
  toggleEmailModal: jest.fn(),
}

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableOpacity')
})

jest.mock('TouchableHighlight', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

describe('<ChangeEmail />', () => {
  afterAll = () => {
    jest.mockRestore()
  }
  it('renders correctly', () => {
    const tree = renderer.create(<ChangeEmail {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
