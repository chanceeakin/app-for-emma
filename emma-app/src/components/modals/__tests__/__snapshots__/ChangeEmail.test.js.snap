// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeEmail /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <ExponentLinearGradient
      colors={
        Array [
          4294960596,
          4293904322,
        ]
      }
      endPoint={
        Array [
          1,
          1,
        ]
      }
      location={
        Array [
          0.25,
          0.4,
          1,
        ]
      }
      locations={null}
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingTop": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 45,
            "textAlign": "center",
          }
        }
      >
        Change Email
      </Text>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="New Email"
          placeholderTextColor="#0D1B2A"
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "gray",
              "color": "#114B5F",
              "height": 40,
              "minWidth": 200,
            }
          }
          value="email"
        />
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
          style={
            Array [
              Object {
                "borderColor": "#0D1B2A",
                "borderRadius": 5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "marginTop": 20,
                "paddingHorizontal": 40,
                "paddingVertical": 15,
              },
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0D1B2A",
                  "fontSize": 23,
                  "fontWeight": "600",
                },
                null,
              ]
            }
          >
            Change Email
          </Text>
        </TouchableOpacity>
      </View>
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[MockFunction]}
        style={
          Object {
            "padding": 15,
          }
        }
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Back
        </Text>
      </TouchableHighlight>
    </ExponentLinearGradient>
  </Modal>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[MockFunction]}
    style={
      Object {
        "backgroundColor": "#D37482",
        "borderRadius": 5,
        "marginTop": 20,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      }
    }
    underlayColor="black"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fafafa",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Change Email
    </Text>
  </TouchableHighlight>
</View>
`;
